### ISSUE TITLE
Search on a recordset has unexpected results

### ISSUE THREAD

**Issue by themreza** [odoo/odoo#84472]
Impacted versions: 14.0, 15.0

Steps to reproduce:

Create and run a scheduled action with the following code:

```python
products = env['product.product'].search([
  ('list_price', '>', 1000)
])
refined_products = products.search([
  ('qty_available', '=', 0)
])
log(str(refined_products))
```

Current behavior:

Rather than using the results of the first search (`products`), it bases the second search on all `product.product` records.

Expected behavior:

Assuming there are enough products to match the search criteria, we should get a few products:
![image](https://user-images.githubusercontent.com/5465057/153635322-cea2af2a-62bc-4d50-9695-5184965341d2.png)

The first search is retrieving the records that have `list_price` greater than `1000`.
The second search should further filter these records and combine the domains.

Video/Screenshot link (optional):

https://user-images.githubusercontent.com/5465057/153636168-fdf5f12e-a20f-47ee-b517-83e74997d698.mp4

Support ticket number submitted via odoo.com/help (optional):

**Comment by themreza** [comment]
Apparently `filtered_domain` is the right method to use. It has exactly the same expected behavior as `records.search`.
It would be nice to mention this somewhere in the documentation, perhaps as a note next to `search`, or as a description for `filtered_domain`.
![image](https://user-images.githubusercontent.com/5465057/153643466-5b3d1085-20db-4190-a9ab-17dbb567d685.png)

Ideally, `search` should behave like `filtered_domain` when passed a recordset.

**Comment by intero-chz** [comment]
Yes it should be more clear in the documentation. Looking into the code it is clear enough, because `search` is decorated by `api.model` which documentation already states enough:

> Decorate a record-style method where self is a recordset, but its contents is not relevant, only the model is.
